MAINT: Add a branch step to the contribution guidelines. (#21)
nipreps/dmriprep: CONTRIBUTING.md

Welcome to the dMRIPrep repository!
We're excited you're here and want to contribute.

> **Imposter's syndrome disclaimer**: We want your help. No, really.

Reviewer: ```suggestion
**Impostor syndrome disclaimer**: We want your help. No, really.
```

Reviewer: Also, add the proper attribution for the Impostor syndrome disclaimer. 

Taken from the original dmriprep:

> 
> The imposter syndrome disclaimer was originally written by
> [Adrienne Lowe](https://github.com/adriennefriend) for a [PyCon
> talk](https://www.youtube.com/watch?v=6Uj746j9Heo), and was
> adapted based on its use in the README file for the [MetPy
> project](https://github.com/Unidata/MetPy).

Author: Added in a footnote.

On GitHub, You'll use [Markdown][markdown] to chat in issues and pull requests.
You can think of Markdown as a few little symbols around your text that will allow GitHub
to render the text with a little bit of formatting.
For example, you could write words as bold (`**bold**`), or in italics (`*italics*`),

> or as a [link][rick_roll] (`[link](https://https://youtu.be/dQw4w9WgXcQ)`) to another webpage.

Reviewer: This one isn't displaying properly.

Reviewer: My mistake. Ignore this comment.

Author: Actually, there are two `https://https://youtu...` - although it does not affect the functionality of the link, it is completely wrong and defeats the purpose of showing how to write links. Fixed in the latest commit.

  
1. **[Clone][link_clone] your forked dMRIPrep repository to your machine/computer.**<br />
  While you can edit files [directly on github][link_githubedit], sometimes the changes
  you want to make will be complex and you will want to use a [text editor][link_texteditor]
  that you have installed on your local machine/computer.
  (One great text editor is [vscode][link_vscode]).<br />  
  In order to work on the code locally, you must clone your forked repository.<br />  
  To keep up with changes in the dMRIPrep repository,
  add the ["upstream" dMRIPrep repository as a remote][link_addremote]
  to your locally cloned repository.  
    ```Shell
    git remote add upstream https://github.com/nipreps/dmriprep.git
    ```
    Make sure to [keep your fork up to date][link_updateupstreamwiki] with the upstream repository.<br />  
    For example, to update your master branch on your local cloned repository:  
      ```Shell
      git fetch upstream
      git checkout master
      git merge upstream/master
      ```

1. **Create a [new branch][link_branches] to develop and maintain the proposed code changes

>   prior a pull-request (see below) is submitted and code changes are reviewed and accepted.**<br />

Reviewer: ```suggestion
  prior to submitting a pull-request (see below) and code changes are reviewed and accepted.**<br />
```

Reviewer: Just a suggestion but maybe get rid of "and code changes are reviewed and accepted"

Author: Thanks, I have simplified this sentence and added one more step for the PR review.

  For example:
    ```Shell
    git fetch upstream  # Always start with an updated upstream
    git checkout -b fix/bug-1222 upstream/master
    ```
    Please consider using appropriate branch names as those listed below, and mind that some of them
    are special (e.g., `doc/` and `docs/`):
      * `fix/<some-identifier>`: for bugfixes
      * `enh/<feature-name>`: for new features
      * `doc/<some-identifier>`: for documentation improvements.
        You should name all your documentation branches with the prefix `doc/` or `docs/`
        as that will preempt triggering the full battery of continuous integration tests.

1. **Make the changes you've discussed, following the [dMRIPrep coding style guide](#dMRIPrep-coding-style-guide).**<br />
  Try to keep the changes focused: it is generally easy to review changes that address one feature or bug at a time.
  It can also be helpful to test your changes locally,

>   using an [dMRIPrep development environment][link_devel].

Reviewer: ```suggestion
  using a [dMRIPrep development environment][link_devel].
```

  Once you are satisfied with your local changes, [add/commit/push them][link_add_commit_push]
  to the branch on your forked repository.

1. **Submit a [pull request][link_pullrequest].**<br />
   A member of the development team will review your changes to confirm
   that they can be merged into the main code base.<br />

>    Pull requests titles should begin with a descriptive prefix

Reviewer: ```suggestion
   Pull request titles should begin with a descriptive prefix
```

  It can also be helpful to test your changes locally,
  using a [dMRIPrep development environment][link_devel].
  Once you are satisfied with your local changes, [add/commit/push them][link_add_commit_push]
  to the branch on your forked repository.

1. **Submit a [pull request][link_pullrequest].**<br />
   A member of the development team will review your changes to confirm
   that they can be merged into the main code base.<br />
   Pull request titles should begin with a descriptive prefix
   (for example, `ENH: Adding Gibbs unringing step`):  
     * `ENH`: enhancements or new features ([example][enh_ex])
     * `FIX`: bug fixes ([example][fix_ex])
     * `TST`: new or updated tests ([example][tst_ex])
     * `DOC`: new or updated documentation ([example][doc_ex])
     * `STY`: style changes ([example][sty_ex])
     * `REF`: refactoring existing code ([example][ref_ex])
     * `CI`: updates to continous integration infrastructure ([example][ci_ex])
     * `MAINT`: general maintenance ([example][maint_ex])
     * For works-in-progress, add the `WIP` tag in addition to the descriptive prefix.
       Pull-requests tagged with `WIP:` will not be merged until the tag is removed.

> 1. **Have your PR reviewed by the developers team, and update your changes accordingly in your branch.**<br />

Reviewer: ```suggestion
1. **Have your PR reviewed by the development team, and update your changes accordingly in your branch.**<br />
```

>    The reviewers will take special care in assisting you address their comments, as well as dealing with conflicts

Reviewer: ```suggestion
   The reviewers will take special care in assisting you to address their comments, as well as dealing with conflicts
```
